shimmingtoolbox/shim/sequencer.py

```diff
@@ -176,55 +174,56 @@ def _eval_static_shim(opt: Optimizer, nii_fieldmap_orig, nii_mask, coef, slices,
         # Save coils
         nii_merged_coils = nib.Nifti1Image(opt.merged_coils, nii_fieldmap_orig.affine, header=nii_fieldmap_orig.header)
         nib.save(nii_merged_coils, os.path.join(path_output, "merged_coils.nii.gz"))
-
     unshimmed = nii_fieldmap_orig.get_fdata()
-
     # If the fieldmap was changed (i.e. only 1 slice) we want to evaluate the output on the original fieldmap
     merged_coils, _ = opt.merge_coils(unshimmed, nii_fieldmap_orig.affine)
-
     # Initialize
     shimmed = np.zeros(unshimmed.shape + (len(slices),))
     corrections = np.zeros(unshimmed.shape + (len(slices),))
     masks_fmap = np.zeros(unshimmed.shape + (len(slices),))
+    list_shim_slice = []
     for i_shim in range(len(slices)):
-        # Calculate shimmed values
-        correction_per_channel = coef[i_shim] * merged_coils
-        corrections[..., i_shim] = np.sum(correction_per_channel, axis=3, keepdims=False)
-        shimmed[..., i_shim] = unshimmed + corrections[..., i_shim]
-
         # Create non binary mask
         masks_fmap[..., i_shim] = resample_mask(nii_mask, nii_fieldmap_orig, slices[i_shim]).get_fdata()
-
-        ma_shimmed = np.ma.array(shimmed[..., i_shim], mask=masks_fmap[..., i_shim]==False)
-        ma_unshimmed = np.ma.array(unshimmed, mask=masks_fmap[..., i_shim]==False)
-        std_shimmed = np.ma.std(ma_shimmed)
-        std_unshimmed = np.ma.std(ma_unshimmed)
-        mae_shimmed = np.ma.mean(np.ma.abs(ma_shimmed))
-        mae_unshimmed = np.ma.mean(np.ma.abs(ma_unshimmed))
-        mse_shimmed = np.ma.mean(np.square(ma_shimmed))
-        mse_unshimmed = np.ma.mean(np.square(ma_unshimmed))
-
-        if opt_criteria is None or opt_criteria == 'mse':
-            if mse_shimmed > mse_unshimmed:
-                logger.warning("Verify the shim parameters. Some give worse results than no shim.\n"
-                               f"i_shim: {i_shim}")
-        elif opt_criteria == 'mae':
-            if mae_shimmed > mae_unshimmed:
-                logger.warning("Verify the shim parameters. Some give worse results than no shim.\n"
-                               f"i_shim: {i_shim}")
-        elif opt_criteria == 'std':
-            if std_shimmed > std_unshimmed:
-                logger.warning("Verify the shim parameters. Some give worse results than no shim.\n"
-                               f"i_shim: {i_shim}")
-
-        logger.debug(f"Slice(s): {slices[i_shim]}\n"
-                     f"MAE:\n"
-                     f"unshimmed: {mae_unshimmed}, shimmed: {mae_shimmed}\n"
-                     f"MSE:\n"
-                     f"unshimmed: {mse_unshimmed}, shimmed: {mse_shimmed}\n"
-                     f"STD:\n"
-                     f"unshimmed: {std_unshimmed}, shimmed: {std_shimmed}"
-                     f"current: \n{coef[i_shim, :]}")
+        # Calculate shimmed values
+        if not np.any(coef[i_shim]):
+            shimmed[..., i_shim] = unshimmed
```

Comment:
```suggestion
            shimmed[..., i_shim] = unshimmed
            continue
```

You can then unindent that whole else block